src/cli.js

```diff
@@ -0,0 +1,168 @@
+    metavar: 'FILE'
+  },
+  cpus: {
+    abbr: 'c',
+    help: '(all by default) Determines the number of processes started.'
+  },
+  verbose: {
+    abbr: 'v',
+    choices: [0, 1, 2],
+    default: 0,
+    help: 'Show more information about the transform process'
+  },
+  dry: {
+    abbr: 'd',
+    flag: true,
+    help: 'Dry run (no changes are made to files)'
+  },
+  print: {
+    abbr: 'p',
+    flag: true,
+    help: 'Print output, useful for development'
+  },
+  babel: {
+    flag: true,
+    default: true,
+    help: 'Apply Babel to transform files'
+  },
+  extensions: {
+    default: 'js',
+    help: 'File extensions the transform file should be applied to'
+  },
+  ignorePattern: {
+    full: 'ignore-pattern',
+    list: true,
+    help: 'Ignore files that match a provided glob expression'
+  },
+  ignoreConfig: {
+    full: 'ignore-config',
+    list: true,
+    help: 'Ignore files if they match patterns sourced from a configuration file (e.g., a .gitignore)',
+    metavar: 'FILE'
+  },
+  runInBand: {
+    flag: true,
+    default: false,
+    full: 'run-in-band',
+    help: 'Run serially in the current process'
+  },
+  silent: {
+    abbr: 's',
+    flag: true,
+    default: false,
+    full: 'silent',
+    help: 'No output'
+  },
+  parser: {
+    choices: ['babel', 'babylon', 'flow'],
+    default: 'babel',
+    full: 'parser',
+    help: 'The parser to use for parsing your source files (babel | babylon | flow)'
+  },
+  version: {
+    flag: true,
+    help: 'print version and exit',
+    callback: function() {
+      return [
+        `jscodeshift: ${pkg.version}`,
+        dependencyVersions(),
+      ].join('\n');
+    },
+  },
+};
+
+function cliGenerator(cliOptions) {
+  const name = cliOptions.name;
+  const packageVersion = cliOptions.packageVersion;
+  const resolve = cliOptions.resolve;
+  const options = cliOptions.options;
+
+  const version = {
+    flag: true,
+    help: 'print version and exit',
+    callback: function() {
+      return [
+        `${name}: ${packageVersion}`,
+        ` - jscodeshift: ${pkg.version}`,
+        dependencyVersions(),
+      ].join('\n');
+    },
+  };
+
+  const transform = {
+    position: 0,
+    help: 'The transformer to use',
+    required: true,
+  };
+
+  const path = {
+    position: 1,
+    help: 'Files or directory to transform',
+    list: true,
+    metavar: 'FILE',
+    required: true,
+  }
+
+  const args = require('nomnom')
+    .script(name)
+    .options(_.assign({}, defaults, {version, transform, path}))
+    .parse();
+
+  const transformerPath = resolve(args.transform);
+  fs.stat(transformerPath, function (error) {
+    if (error) {
+      console.error(`The transform ${args.transform} does not exist.`);
+      process.exit(1);
+    }
+
+    console.log(`Using transform ${args.transform}.`);
+
+    Runner.run(
+      transformerPath,
+      args.path,
+      _.assign({}, options, args)
```

Comment:
`jscodeshift` [only supports node 4 and up](https://github.com/facebook/jscodeshift/blob/master/package.json#L21), so you could just use the builtin `Object.assign` here